File: app/src/test/java/com/example/calculator/MainActivityTest.java
package com.example.calculator;

import static org.junit.Assert.*;

import org.junit.Test;

public class MainActivityTest {

    @Test
    public void calcularPotencia() {
        int result = MainActivity.calcularPotencia(2, 3);
        assertEquals(8, result);
    }

    @Test
    public void calcularS() {
        double result = MainActivity.calcularsuma(5.2, 3);
        assertEquals(8.2, result, 0.001); // Allow for small floating-point differences
    }

    @Test
    public void calcularR() {
        double result = MainActivity.calcularresta(5, 3);
        assertEquals(2, result, 0.001);
    }

    @Test
    public void calcularM() {
        double result = MainActivity.calcularmulti(5, 3);
        assertEquals(15, result, 0.001);
    }

    @Test
    public void calcularD() {
        double result = MainActivity.calculardiv(6, 3);
        assertEquals(2, result, 0.001);
    }
}

File: app/src/test/java/com/example/calculator/MainActivityTest.kt
package com.example.calculator

import com.example.calculator.MainActivity.Companion.calcularPotencia
import com.example.calculator.MainActivity.Companion.calculardiv
import com.example.calculator.MainActivity.Companion.calcularmulti
import com.example.calculator.MainActivity.Companion.calcularresta
import com.example.calculator.MainActivity.Companion.calcularsuma
import org.junit.Assert
import org.junit.Test

class MainActivityTest {
    @Test
    fun calcularPotencia() {
        val result = calcularPotencia(2, 3)
        Assert.assertEquals(8, result.toLong())
    }

    @Test
    fun calcularS() {
        val result = calcularsuma(5.2, 3.0)
        Assert.assertEquals(8.2, result, 0.001) // Allow for small floating-point differences
    }

    @Test
    fun calcularR() {
        val result = calcularresta(5.0, 3.0)
        Assert.assertEquals(2.0, result, 0.001)
    }

    @Test
    fun calcularM() {
        val result = calcularmulti(5.0, 3.0)
        Assert.assertEquals(15.0, result, 0.001)
    }

    @Test
    fun calcularD() {
        val result = calculardiv(6.0, 3.0)
        Assert.assertEquals(2.0, result, 0.001)
    }
}